lib/auth/auth.go

```diff
@@ -4329,6 +4329,18 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 		return req, nil
 	}
 
+	// calculate the promotions
+	reqCopy := req.Copy()
+	promotions, err := modules.GetModules().GenerateAccessRequestPromotions(ctx, a.Services, reqCopy)
+	if err != nil {
+		// Do not fail the request if the promotions failed to generate.
+		// The request promotion will be blocked, but the request can still be approved.
+		log.WithError(err).Warn("Failed to generate access list promotions.")
+	}
+
+	// update the request with additional reviewers if possible.
+	updateAccessRequestWithAdditionalReviewers(ctx, req, a.AccessListClient(), promotions)
```

Comment:
Why not move this inside `if promotions != nil {` below? I think it's cleaner to process a promotion when you know it exists instead of checking inside the function and a few lines below.

Reply:
I did this specifically to ensure that the suggested reviewers is modified during the creation of the access request -- without doing this here, we can't update the suggested reviewers in the access request itself without doing a subsequent update after the fact.